TFEngine.py

```diff
@@ -1344,48 +1345,81 @@ def _do_save(self):
         return False
     return True
 
-  def eval_model(self, output_file=None, output_per_seq_file=None):
+  def eval_model(self, output_file=None, output_per_seq_file=None, loss_name=None, output_per_seq_format=["seq_len", "score", "pos_score"]):
     """
     Eval the current model on the eval datasets (dev + eval, whatever is set).
     See also :func:`self.search` for performing beam search.
 
     :param str|None output_file: if given, will save the results to this file (total err/score for each dataset)
     :param str|None output_per_seq_file: if given, will save the err/score for each sequence
+    :param str|None loss_name: specifies the loss which will be written to output_file
+    :param list[str]|tuple[str]|None output_per_seq_format:
```

Comment:
Your default in the config is not a list or tuple, but just a `str`. Maybe you also want to allow that here.